### PR TITLE
[Stripe testing] Replaces stubs for VCR calls on profile_storer_spec.rb

### LINKED ISSUE #12164: Replace stubs on ./spec/lib/stripe/profile_storer_spec.rb
### What we should change and why (this is tech debt)

Relates to #11890.

### Impact and timeline

Replacing all Stripe stubs with real VCR/HTTP requests.

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Closes #12164 .
Continues #11890 

- Replaces stubs for VCR calls on profile_storer_spec.rb
- Removes an unnecessary environment variable (STRIPE_ACCOUNT) from a previously merged spec.  

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by rioug** [comment]
@filipefurtad0 This PR upgrade sidequick, is it intended ? also there is nothing about replacing stubs for VCR calls on profile_storer_spec.rb

**Comment by filipefurtad0** [comment]
Oh no.. I must of messed up badly with the last rebase... :-(
I'll try to recover it...

Sorry, moving back to in dev.

**Comment by filipefurtad0** [comment]
Ok, should be good now @rioug .

It seems I've rebased it badly and did not notice my mistake.  Moved to code review.

File: spec/lib/stripe/profile_storer_spec.rb [hunk 1]
```diff
@@ -7,45 +7,66 @@ module Stripe
     include StripeStubs
```
Comment (rioug) [inline spec/lib/stripe/profile_storer_spec.rb:7]: You should be able to remove this now

**Review by rioug — APPROVED** [review]
Looks good ! I just spotted a small clean up, see comment.

File: spec/lib/stripe/profile_storer_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/lib/stripe/profile_storer_spec.rb:7]: Yes, nice catch. Removed and squashed.

**Review by dacook — APPROVED** [review]
Yay, more real-world tests!

**Pull request merged**